packages/expo-speech/src/Speech/Speech.types.ts

```diff
@@ -45,6 +46,10 @@ export type SpeechOptions = {
    */
   voice?: string;
   _voiceIndex?: number;
+
+  /**
+   * A callback that is invoked when the spoken utterance reaches a word. Android and iOS only pass the charIndex and charLength of `SpeechSynthesisEvent`.
```

Comment:
Also, correct me if I'm wrong but web [`SpeechSynthesisEvent`](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisEvent) doesn't have the `charLength` property so this would cause typescript errors.

I'm wondering if having both web `SpeechEventCallback` and your previous callback `({ charIndex:  number, charLength: number }) => void` a union type would make sense:
```ts
/**
 * Your description here from previous commits
 * @platform ios
 * @platform android
 */
export type NativeBoundaryEventCallback = (({ charIndex:  number, charLength: number }) => void;

// then in options
onBoundary?: NativeBoundaryEventCallback | SpeechEventCallback | null;
```
There are already union types for other options like `onStart`. What do you think? 🤔

Reply:
> web [SpeechSynthesisEvent](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisEvent) doesn't have the charLength property

yes, I mentioned this in https://github.com/expo/expo/pull/20726#discussion_r1068467834, and that it "does exist in this [type](https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L1717-L1723) that my IDE is able to find when I "go to definition" (a RN shim?)." know what that type is about?

yes that type change makes sense. i'll do that

Reply:
I have no idea where this type comes from either 🤷 but I'd stick to the MDN docs where `charLength` isn't mentioned

Reply:
Added commits with the union type and ran `yarn lint --fix`